### ISSUE TITLE
'<' not supported between instances of 'float' and 'complex'

### ISSUE THREAD

**Issue by jinyi12** [facebookresearch/dadaptation#10]
Hi, there is an issue with the optimizer at d = max(d, min(d_hat, d*growth_rate)) when d_hat is complex.
In addition, will the package benefit from a correction such as that in [here](https://github.com/pytorch/pytorch/issues/59998)

**Comment by adefazio** [comment]
I will look into this, should be easy to fix. Do you have a suggestion of some code I can run to test if it works? any open source repo optimizing over complex tensors?

**Comment by jinyi12** [comment]
Hi @adefazio, thanks for the prompt reply! You can try out [Geo-FNO](https://github.com/neural-operator/Geo-FNO), looking into the /elasticity/elas-geofno.py file
The data could be obtained from [Geo-FNO Dataset](https://drive.google.com/drive/folders/1YBuaoTdOSr_qzaow-G-iwvbUI7fiUzu8?usp=sharing). The data used should be in /Geo-FNO/elasticity/Meshes, just replace the path to these meshes in the .py file, and it should be able to run.

Best,
Jin

**Comment by adefazio** [comment]
Please let me know if you have any issues. For that particular problem/dataset the default growth_rate is too high, I found using 1.2 worked best. Any kind of learning rate warmup would also help, DAdaptation can be unstable on some problems at the beginning of optimization.

**Comment by jinyi12** [comment]
Hi @adefazio, thanks for adding complex support. However for my own dataset, I am getting the same error at 
`d = max(d, min(d_hat, d*growth_rate))` in `dadapt_adam.py`. Upon inspecting using the debugger, I found that `d_hat` is complex valued, hence the error.
